library-valcon-internship/src/components/Modals/ModalAddAuthor/ModalAddAuthor.tsx

```diff
@@ -0,0 +1,124 @@
+import { Dispatch, FormEvent, SetStateAction, SyntheticEvent, useState } from 'react'
+
+import ReactDOM from 'react-dom'
+
+import { addNewAuthor } from '../../../services/AuthorService'
+import '../Modals.css'
+import './ModalAddAuthor.css'
+
+interface AddAuthorProps {
+  show: boolean,
+  closeModal: () => void,
+  retrieveAuthors: boolean,
+  setRetrieveAuthors: Dispatch<SetStateAction<boolean>>
+}
+
+const ModalAddAuthor = ({ show, closeModal, retrieveAuthors, setRetrieveAuthors }: AddAuthorProps) => {
+  const [ firstName, setFirstName ] = useState('')
+  const [ lastName, setLastName ] = useState('')
+  const [ invalidFirstName, setInvalidFirstName ] = useState(false)
+  const [ invalidLastName, setInvalidLastName ] = useState(false)
+  const [ invalidAuthorData, setInvalidAuthorData ] = useState(false)
+  if (!show) return null
+  const handleOnChangeFirstName = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
+    setFirstName(currentTarget.value)
+  }
+  const handleOnBlurFirstName = () => {
+    if (firstName.trim() === '')
+      setInvalidFirstName(true)
+  }
+  const handleOnChangeLastName = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
+    setLastName(currentTarget.value)
+  }
+  const handleOnBlurLastName = () => {
+    if (lastName.trim() === '')
+      setInvalidLastName(true)
+  }
+  const handleOnSubmit = (e: SyntheticEvent) => {
+    e.preventDefault()
+    if (firstName.trim() === '') {
+      setInvalidFirstName(true)
+      return
+    }
+    if (lastName.trim() === '') {
+      setInvalidLastName(true)
+      return
+    }
```

Comment:
Wouldn't it be better if we trim on setting the name, so we don't send extra spaces on server?